Fix incorrect tests and override canonical associate

cxzhong opened the pull request:
Because one use roundtieeven, one use roundawayzero. They are not equal when half integers
And also move https://github.com/passagemath/passagemath/pull/1770 into this file
<!-- ^ Please provide a concise and informative title. -->
<!-- ^ Don't put issue numbers in the title, do this in the PR description below. -->
<!-- ^ For example, instead of "Fixes #12345" use "Introduce new method to calculate 1 + 2". -->
<!-- v Describe your changes below in detail. -->
<!-- v Why is this change required? What problem does it solve? -->
<!-- v If this PR resolves an open issue, please link to it here. For example, "Fixes #12345". -->



### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [ ] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example, -->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->

cxzhong committed:
Remove incorrect tests

Because one use roundtieeven, one use roundawayzero

cxzhong committed:
Remove unused variable in quadratic field tests

Removed unused variable 'b' from test loop.

nbruin commented on src/sage/rings/number_field/number_field_element_quadratic.pyx:
```diff
@@ -2407,13 +2407,9 @@ cdef class NumberFieldElement_quadratic(NumberFieldElement_absolute):
             ....:    assert a.round() == round(K2(a)), a
             ....:    assert a.round() == round(K3(a)), a
             ....:    assert a.round() == round(K5(a)), a
-            ....:    assert round(a+b*sqrt(2.)) == round(a+b*sqrt2), (a, b)
```
Perhaps:
```
        #outside of rounding half integers, rounding between quadratic numbers are
        #floats should agree (assuming enough precision is available)
        if b != 0 or a+1/2 in ZZ:
            assert round(a+b*sqrt(2.)) == round(a+b*sqrt2), (a,b)
            [etc.]
```
It's not unreasonable to test it and apparently the authors thought some stress testing is warranted (and from this report we've already seen that these tests indeed detect delicate behaviour!). So I'd rather maintain the tests in general and just avoid the edge cases where rounding doesn't agree.

cxzhong committed:
Enhance rounding tests and add canonical associate method

Added tests for rounding behavior with quadratic fields and implemented a canonical associate method.
